src/services/produce_api/produceAPI/produce/tests.py

```diff
@@ -1,15 +1,39 @@
-from django.test import TestCase
-from rest_framework.test import APIRequestFactory
+import json
 
-from produce.models import Produce
-from produce.serializers import ProduceSerialiser
+from django.urls import reverse
+from django.contrib.auth.models import User
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+from rest_framework.test import APITestCase, RequestsClient
 
-#using the standard API Request Factory to test the create request
-createTest = APIRequestFactory()
-request = createTest.post('/products/', {'name': 'testname'})
-print('created test ran successifully')
+from .models import Produce
+from .serializers import ProduceSerialiser
 
-#using the standard Request Factory to test the update request
-updateTest = APIRequestFactory()
-request = updateTest.put('/products/100/', {'name': 'updating test name'})
-print('updated succesifully')
+client = RequestsClient()
+response = client.get('http://127.0.0.1:8000/products/')
+assert response.status_code == 200
+
+class CreateProductTestCase(APITestCase):
+    def test_new_productcreation(self):
+        data = {"name": "nakati", "description": "this is a test description"}
+        response = self.client.post('/products/', data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_updateProduct(self):
+        data = {"name": "testchange"}
+        response = self.client.put('/products/2/', data)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+class CredentialsTestCase(APITestCase):
+    test_url = reverse('prod')
+    def setup(self):
+        self.user = User.objects.create_user(username='testuser', email='testuser@farmrail.com', password='verystrongpsw')
```

Comment:
refactor line by placing valuables on new lines

Reply:
not applicable in python